packages/tldraw/src/state/commands/duplicatePage/duplicatePage.ts

```diff
@@ -9,23 +9,23 @@ export function duplicatePage(app: TldrawApp, pageId: string): TldrawCommand {
   } = app
 
   const page = app.document.pages[pageId]
-
-  const newId = Utils.uniqueId()
+  const newIds = Object.fromEntries(
+    Object.keys(page.shapes).map(id => [id, Utils.uniqueId()])
+  )
 
   const nextPage = {
     ...page,
-    id: newId,
+    id: newIds[page.id],
     name: page.name + ' Copy',
     shapes: Object.fromEntries(
```

Comment:
We'd also need to duplicate any bindings, create references to the new shape ids, and update references to bindings in shape's handles. IIRC we solved this in the new beta by essentially copying and pasting the content from one page to the next, since all of this logic is already present in getContent / putContent.

Reply:
Tried to grok what `getContent` and `insertContent` do; tell me if I missed anything.